### ISSUE TITLE
Provide Output Routines 'Image'

### ISSUE THREAD

**Issue by RREE** [onox/json-ada#4]
The attached patch adds Image functions to all JSON objects. The resulting output is terse but valid JSON.

[Added-new-functions-Image-for-JSON-output.zip](https://github.com/onox/json-ada/files/1648697/Added-new-functions-Image-for-JSON-output.zip)

**Comment by onox** [comment]
Thanks for providing a patch (including some unit tests 👍)! I have a few remarks:

1. I don't mind the use of your nickname in your copyright notices, but can you add a valid e-mail address (in case you ever need to be contacted)?

2. Place the `is` keyword on the previous line (I generally only place them on a new line if there are declarations in the procedure/function _and_ if the parameters are on separate lines (to prevent a line becoming too long)

3. In `function Image (Object : JSON_String_value) return String`:

- Indent the `when` statements and place the `Append` calls on a new line
- Prefix them with `SU.` instead of using `use Ada.Strings.Unbounded` (for consistency with the rest of the code)
- Furthermore, I think you can avoid defining the `Escape` function by just writing `Text  : Unbounded_String renames Object.String_Value` in the declarative part of the `Image` function (or use `Object.String_Value` directly)
- Rename `Idx` to `Index`
- Use of `ASCII` results in a compile error: `violation of restriction "NO_OBSOLESCENT_FEATURES"`. You can use `use Ada.Characters.Latin_1` and simply remove the `ASCII.` parts.

4. In the other `Image` functions: can you rename `Img` to `Result` and replace `2` with `Result'First + 1`

5. Rewrite `function Image (Object : JSON_Boolean_value) return String` to:

```ada
function Image (Object : JSON_Boolean_value) return String is
  (if Object.Boolean_Value then "true" else "false");
```

6. Rewrite `function Image (Object : JSON_Null_value) return String` to:

```ada
function Image (Object : JSON_Null_value) return String is ("null");
```

**Comment by RREE** [comment]
Hi,

I applied all your proposed changes. I'm still convinced that '2' is more readable than Result'First + 1, but never mind.

best regards
   Rolf
[applied-naming-and-layout-changes-as-proposed-by-onox.zip](https://github.com/onox/json-ada/files/1676339/applied-naming-and-layout-changes-as-proposed-by-onox.zip)

**Comment by onox** [comment]
Thanks. After some thinking I've adjusted the patch to use '2' like in your original patch and moved some `is` keywords around.

Fixed in cb0df4a
